e2e/migrations/migrations_test.go

```diff
@@ -0,0 +1,176 @@
+package e2e
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func isSuccess(execCmd *exec.Cmd, retries int, keyword string) (bool, string) {
+	numAttempts := retries
+	success := false
+	var out bytes.Buffer
+
+	for numAttempts > 0 {
+		time.Sleep(2 * time.Second)
+
+		cmd := *execCmd
+		cmd.Stdout = &out
+
+		err := cmd.Run()
+		if err != nil {
+			return false, ""
+		}
+
+		success = (out.String() != "" && !strings.Contains(out.String(), keyword))
+		if success {
+			return success, out.String()
+		}
+
+		out.Reset()
+		numAttempts -= 1
+	}
+
+	return success, out.String()
+}
+
+func allNomadNodesAreReady(retries int) (bool, string) {
+	cmd := exec.Command("nomad", "node-status")
+	return isSuccess(cmd, retries, "initializing")
```

Comment:
Shouldn't this be checking `ready`?

Reply:
The problem with checking `ready` is that one node might be ready, while the others are not. So by checking that `initializing` is not in the output, we have a better sense that all nodes are actually ready.

Alternatively, we could count the number of times `ready` appears and check that it is equal to the number of clients we are running.